src/api/transactions.js

```diff
@@ -1,5 +1,56 @@
 const express = require('express');
+const { Transaction } = require('../db/models');
+const { getDate } = require('../utils');
 
 const router = express.Router();
 
+/**
+ * Get transactions grouped by date, month, or year between two intervals 
+ * Sample query: /api/transactions/intervals?orderBy=day&startDate=2021-04&endDate=2022-09
+ */
+ router.get('/intervals', async (req, res, next) => {
+    try {
+      if (!req.user) {
+        return res.status(401).json({ error: 'Please log in' });
+      }
+      const { startDate, endDate, orderBy } = req.query;
+  
+      if (!startDate || !endDate){
+        return res.status(400).json({ error: 'Mising start or end date' });
+      }
+  
+      const isValidStartDate = Date.parse(startDate); 
+      const isValidEndDate = Date.parse(endDate); 
+  
+      if (isNaN(isValidStartDate) || isNaN(isValidEndDate)){
+        return res.status(400).json({ error: 'Invalid date. Please input date in yyyy-mm-dd format' });
+      }
+  
+      const validOrderByOptions = ['day', 'month', 'year'];
+      if (orderBy && !validOrderByOptions.includes(orderBy)){
+        return res.status(400).json({ error: 'Invalid order by option' });
+      }
+  
+      // Default value: group by date
+      const orderByParam = orderBy || "day";
+  
+      const t = {}
```

Comment:
A more meaningful variable name could be used here, especially since t could be interpreted several ways in the context of this route (for example, total or transaction)